### ISSUE TITLE
Schema types and TypeScript

### ISSUE THREAD

**Issue by UncleVic** [Automattic/mongoose#10027]
Doesn't work boolean and ObjectId type
For example

```
interface ITransaction {
  orderId: Schema.Types.ObjectId;
  confirm: boolean;
}

interface ITransactionDocument extends ITransaction, Document {
  order?: IOrderDocument;
}

export type ITransactionStatic = {
  createFromData(params: TTransctionCreate): Promise<ITransactionDocument>;
  findByOrderId(id: Schema.Types.ObjectId): Promise<ITransactionDocument[]>;
};

export type ITransactionStatic = {
  findByOrderId(id: Schema.Types.ObjectId): Promise<ITransactionDocument[]>;
};

export type ITransactionModel = ITransactionStatic & Model<ITransactionDocument>;

const TransactionSchema = new Schema<ITransactionDocument, ITransactionModel, ITransaction>(
  {
    orderId: { type: Schema.Types.ObjectId, ref: 'Orders' },
    confirm: {
      type: Schema.Types.Boolean,
      required: true,
      default: true,
    },
});
```
And I have 2 errors
```
Type '{ type: typeof Schema.Types.ObjectId; ref: string; }' is not assignable to type 'typeof SchemaType | Function[] | SchemaTypeOptions<ObjectId> | Schema<Document<any, {}>, Model<any, any>, undefined> | ... 4 more ... | undefined'.
  Types of property 'type' are incompatible.
    Type 'typeof ObjectId' is missing the following properties from type 'ObjectId': auto, default, immutable, index, and 8 more.
    
Type '{ type: typeof Schema.Types.Boolean; required: true; default: true; }' is not assignable to type 'boolean | typeof SchemaType | Function[] | SchemaTypeOptions<false> | Schema<Document<any, {}>, Model<any, any>, undefined> | ... 8 more ... | undefined'.
  Types of property 'type' are incompatible.
    Type 'typeof Boolean' is not assignable to type 'boolean | undefined'.
      Type 'typeof Boolean' is not assignable to type 'true'
```

**Comment by IslandRhythms** [comment]
what is `IOrderDocument`?

**Comment by UncleVic** [comment]
> what is `IOrderDocument`?

This is virtual function. But it does not matter here.

**Comment by vkarpov15** [comment]
Should be fixed by #10030 but I'll need to double check about `Schema.Types.Boolean`

**Comment by vkarpov15** [comment]
Confirmed, the fix from #10030 works with `Schema.Types.Boolean`. Fix will be in 5.12.2 :+1:
